lib/wasix/src/syscalls/mod.rs

```diff
@@ -1481,14 +1484,53 @@ where
 }
 
 // Function to prepare the WASI environment
-pub(crate) fn _prepare_wasi(wasi_env: &mut WasiEnv, args: Option<Vec<String>>) {
+pub(crate) fn _prepare_wasi(
+    wasi_env: &mut WasiEnv,
+    args: Option<Vec<String>>,
+    envs: Option<Vec<(String, String)>>,
+) {
     // Swap out the arguments with the new ones
     if let Some(args) = args {
         let mut wasi_state = wasi_env.state.fork();
         wasi_state.args = args;
         wasi_env.state = Arc::new(wasi_state);
     }
 
+    // Update the env vars
+    if let Some(envs) = envs {
+        let mut guard = wasi_env.state.envs.lock().unwrap();
+
+        let mut existing_envs = guard
+            .iter()
+            .map(|b| {
+                let string = String::from_utf8_lossy(b);
+                let (key, val) = string.split_once('=').expect("env var is malformed");
+
+                (key.to_string(), val.to_string().as_bytes().to_vec())
+            })
+            .collect::<Vec<_>>();
+
+        'outer: for (key, val) in envs {
+            let val = val.as_bytes().to_vec();
+
+            for (existing_key, existing_val) in existing_envs.iter_mut() {
+                if existing_key == &key {
+                    *existing_val = val;
+
+                    continue 'outer;
+                }
+            }
+
+            existing_envs.push((key, val));
+        }
```

Comment:
I think this may be a bit more "Rustonic". I written on the fly, so there might be some corrections to do

```suggestion
        for (key, val) in envs {
            let val = val.as_bytes().to_vec();
            match existing_envs.iter_mut().find(|(existing_key, existing_val)| existing_key == &key) {
                Some((_, existing_val)) =>  *existing_val = val,
                None => existing_envs.push((key, val))
            }
        }
```